R/FilterStateExpr.R

```diff
@@ -0,0 +1,350 @@
+
+#' @name FilterStateExpr
+#' @docType class
+#'
+#'
+#' @title `FilterStateExpr` Class
+#'
+#' @description Class to handle filter expression.
+#'
+#'
+#' @details
+#' This class is responsible for displaying filter card and returning filter expression
+#'
+#' @keywords internal
+#'
+FilterStateExpr <- R6::R6Class( # nolint
+  classname = "FilterStateExpr",
+  # public methods ----
+  public = list(
+    #' @description
+    #' Initialize a `FilterStateExpr` object
+    #' @param id (`character(1)`)\cr
+    #'   identifier of the filter
+    #' @param title (`reactive`)\cr
+    #'   title of the filter
```

Comment:
Seems like this should be `character` but perhaps it should be stored in a `reactiveVal`?

Reply:
good point, I'll fix it